rotkehlchen/tasks/assets.py

```diff
@@ -74,3 +108,116 @@ def autodetect_spam_assets_in_db(user_db: DBHandler) -> None:
                 'blockchain': chain.to_blockchain().serialize(),
             },
         )
+
+
+def augmented_spam_detection(user_db: DBHandler) -> None:
+    """
+    Make a more exhaustive check looking for spam tokens. What it does is:
+    1. Query the assets in the history events
+    2. Filter the assets that have only receive events
+    3. Check if the logs for the tx receipt has more than MULTISEND_SPAM_THRESHOLD transfers
+    4. Check price using defillama as oracle
+
+    For any asset that passes all the previous filters we mark it as spam and it is ignored.
+    """
+    history_db = DBHistoryEvents(user_db)
+    evm_tx_db = DBEvmTx(user_db)
+    inquirer = Inquirer()
+    globaldb = GlobalDBHandler()
+    usd_asset = A_USD.resolve_to_fiat_asset()
+    chains_with_spam: set[ChainID] = set()
+
+    spam_assets: set[str] = set()
+    with user_db.conn.read_ctx() as cursor:
+        cursor.execute('SELECT DISTINCT asset FROM history_events')
+        for (asset_id,) in cursor:
+            asset = Asset(asset_id)
+            if (
+                not asset.is_evm_token() or
+                asset.identifier in spam_assets
+            ):
+                continue
+
+            events: list['EvmEvent'] = history_db.get_history_events(
+                cursor=cursor,
+                filter_query=EvmEventFilterQuery.make(assets=(asset,)),
+                has_premium=True,
+                group_by_event_ids=False,
+            )
+
+            # check if all the events are receive and nothing else
+            if (
+                len(events) == 0 or
+                not all(event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE for event in events)  # noqa: E501
+            ):
+                log.debug(
+                    f'Skipping {asset=} during spam detection because not '
+                    'all the events are receive',
+                )
+                continue
+
+            # check if the transaction could be a multisend
+            receipt = evm_tx_db.get_receipt(
+                cursor=cursor,
+                tx_hash=events[0].tx_hash,
+                chain_id=ChainID(events[0].location.to_chain_id()),
+            )
+            if receipt is None:
+                log.info(f'Could not find receipt for event {events[0].identifier=}')
+                continue
+
+            transfer_counter = 0
+            for log_event in receipt.logs:
+                if log_event.topics[0] == ERC20_OR_ERC721_TRANSFER:
+                    transfer_counter += 1
+
+                if transfer_counter == MULTISEND_SPAM_THRESHOLD:
+                    break
+            else:
+                log.debug(f'Transaction {events[0].tx_hash=} was not detected as a spam transaction')  # noqa: E501
+                continue
```

Comment:
Dude this is too much logs spam. Please don't do this!

```suggestion
                continue  # not spam
```

Reply:
It doesn't happen for every event log. Is just after executing them. Anyway removed it

Reply:
It's an unecessary spamming log entry